gui/src/main.rs

```diff
@@ -334,7 +334,12 @@ impl Config {
                 Err(ConfigError::NotFound) => Ok(Config::Install(datadir_path, network)),
                 Err(e) => Err(format!("Failed to read configuration file: {}", e).into()),
             }
-        } else if !datadir_path.exists() {
+        } else if !datadir_path.exists()
+            | (!datadir_path.join("bitcoin").exists()
+                & !datadir_path.join("testnet").exists()
+                & !datadir_path.join("signet").exists()
+                & !datadir_path.join("regtest").exists())
```

Comment:
Why binary operators?

```suggestion
        } else if !datadir_path.exists()
            || (!datadir_path.join("bitcoin").exists()
                && !datadir_path.join("testnet").exists()
                && !datadir_path.join("signet").exists()
                && !datadir_path.join("regtest").exists())
```

Reply:
missing some cafeine i guess